front/src/modules/simulationResult/components/ManchetteWithSpaceTimeChart/ManchetteWithSpaceTimeChart.tsx

```diff
@@ -499,13 +473,18 @@ const ManchetteWithSpaceTimeChartWrapper = ({
       const hoveredTrainId = hoveredItem.element.pathId;
       if (!isTrainId(hoveredTrainId)) return;
       const train = projectedTrains.find((projectedTrain) => projectedTrain.id === hoveredTrainId);
-      if (train) {
-        setDraggingState({
-          draggedTrain: train,
-          initialDepartureTime: train.departureTime,
-        });
-      } else {
+
+      if (!train) {
         console.error(`No train found with id ${hoveredTrainId}`);
+      } else {
+        const isStartTimeException =
+          isIndividualOccurrenceProjection(train) && !!train.exception?.start_time;
+        if (!isStartTimeException) {
+          setDraggingState({
+            draggedTrain: train,
+            initialDepartureTime: train.departureTime,
+          });
+        }
       }
```

Comment:
```suggestion
const hoveredTrainId = hoveredItem.element.pathId;
  if (!isTrainId(hoveredTrainId)) return;

  const train = projectedTrains.find((t) => t.id === hoveredTrainId);
  if (!train) {
    console.error(`No train found with id ${hoveredTrainId}`);
    return;
  }

  const isStartTimeException =
    isIndividualOccurrenceProjection(train) && !!train.exception?.start_time;

  if (isStartTimeException) return;

  setDraggingState({
    draggedTrain: train,
    initialDepartureTime: train.departureTime,
  });
```

To improve clarity and maintainability, I suggest avoiding deeply nested if statements and keeping the logic as flat as possible. What do you think ?

Reply:
Done :white_check_mark: